mechafil/data.py

```diff
@@ -4,198 +4,108 @@
 import datetime
 from typing import Tuple
 
-EXBI = 2**60
-PIB = 2**50
-
+from . import data_spacescope, data_starboard
+DEFAULT_DATA_BACKEND = 'spacescope'
 
 def get_historical_network_stats(
-    start_date: datetime.date, current_date: datetime.date, end_date: datetime.date
+    start_date: datetime.date, 
+    current_date: datetime.date, 
+    end_date: datetime.date,
+    data_backend: str = DEFAULT_DATA_BACKEND
 ) -> pd.DataFrame:
-    power_df = query_starboard_power_stats(start_date, current_date)
-    onboards_df = query_starboard_daily_power_onboarded(start_date, current_date)
-    stats_df = query_starboard_supply_stats(start_date, current_date)
-    stats_df = stats_df.merge(power_df, on="date", how="inner").merge(
-        onboards_df, on="date", how="inner"
-    )
-    rb_renewal_rate, day_renewed_qa_power = get_day_renewed_power_stats(
-        start_date, current_date, end_date
-    )
-    stats_df["rb_renewal_rate"] = rb_renewal_rate
-    stats_df["day_renewed_qa_power_pib"] = day_renewed_qa_power
-    return stats_df
+    if data_backend.lower() == 'spacescope':
+        return data_spacescope.get_historical_network_stats(start_date, current_date, end_date)
+    elif data_backend.lower() == 'starboard':
+        return data_starboard.get_historical_network_stats(start_date, current_date, end_date)
+    else:
+        raise ValueError("Data Backend: %s not supported!" % (data_backend,))
 
 
 def get_sector_expiration_stats(
     start_date: datetime.date,
     current_date: datetime.date,
     end_date: datetime.date,
+    data_backend: str = DEFAULT_DATA_BACKEND
 ) -> pd.DataFrame:
-    scheduled_df = query_starboard_sector_expirations(start_date, end_date)
-    filter_scheduled_df = scheduled_df[
-        scheduled_df["date"] >= pd.to_datetime(current_date, utc="UTC")
-    ]
-    rbp_expire_vec = filter_scheduled_df["total_rb"].values
-    qap_expire_vec = filter_scheduled_df["total_qa"].values
-    # we need the entire history of known_scheduled_pledge_release, so get the
-    # data from the entire time-window, not just from current-date onwards
-    pledge_release_vec = scheduled_df["total_pledge"].values
-    return rbp_expire_vec, qap_expire_vec, pledge_release_vec
-
+    if data_backend.lower() == 'spacescope':
+        return data_spacescope.get_sector_expiration_stats(start_date, current_date, end_date)
+    elif data_backend.lower() == 'starboard':
+        return data_starboard.get_sector_expiration_stats(start_date, current_date, end_date)
+    else:
+        raise ValueError("Data Backend: %s not supported!" % (data_backend,))
 
 def get_day_renewed_power_stats(
```

Comment:
`data.py` can be greatly simplified. I don't think we should have function here that are not used in other modules. The only functions that are used elsewhere are the following:
* `get_historical_network_stats` -> used in `sim.py`
* `get_sector_expiration_stats` -> used in `sim.py`
*  `query_supply_stats` ->  used in `data.py`
*  `get_storage_baseline_value` -> used in `minting.py`
*  `get_cum_capped_rb_power` -> used in `minting.py`
* `get_vested_amount` -> used in `vesting.py`
* `query_historical_baseline_power` -> used in `data.py`
* `query_historical_rb_power` -> used in `data.py`

The other functions can be removed. Note that `get_cum_capped_qa_power` is being imported in `minting.py`, but it is not used there. Maybe we should delete this import and then delete the function def. If we do this, we can also delete `query_historical_qa_power`.

Reply:
I do use `get_cum_capped_qa_power` in a Jupyter notebook, and some of the other functions that are not listed here are used as well, especially when debugging.  I can understand where you are coming from in terms of keeping mechaFIL clean and only having the necessary functionality in there.  If we want to keep it that way, a suggestion might be to move all data download functionality into its own repository and import it in mechaFIL.  However, I am in support of having a rich python python data download library to 1) quickly debug things, and 2) quickly add new features/simulations.

Let me know how you want to proceed.

Reply:
Got it. If you find it useful, then I am happy to keep it.